File: Api/src/test/java/net/binggl/mydms/test/senders/SenderModelTest.java
package net.binggl.mydms.test.senders;

import static io.dropwizard.testing.FixtureHelpers.fixture;
import static org.assertj.core.api.Assertions.assertThat;

import org.junit.Test;

import com.fasterxml.jackson.databind.ObjectMapper;

import io.dropwizard.jackson.Jackson;
import net.binggl.mydms.senders.Sender;

public class SenderModelTest {

	private static final ObjectMapper MAPPER = Jackson.newObjectMapper();

    @Test
    public void serializesToJSON() throws Exception {
        final Sender sender = new Sender(1, "sender1");

        final String expected = MAPPER.writeValueAsString(
                MAPPER.readValue(fixture("fixtures/sender.json"), Sender.class));

        assertThat(MAPPER.writeValueAsString(sender)).isEqualTo(expected);
    }
    
    @Test
    public void deserializesFromJSON() throws Exception {
        final Sender sender = new Sender(1, "sender1");
        assertThat(MAPPER.readValue(fixture("fixtures/sender.json"), Sender.class))
                .isEqualTo(sender);
    }
}

File: Api/src/test/kotlin/net/binggl/mydms/features/senders/SenderModelTest.kt
package net.binggl.mydms.features.senders

import FixtureHelpers
import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
import net.binggl.mydms.features.senders.Sender
import org.junit.Assert
import org.junit.Test

class SenderModelTest {

    @Test
    fun serializeToJson() {
        val tag = Sender(id = 1, name = "sender1")
        val expected = MAPPER.writeValueAsString(MAPPER.readValue(FixtureHelpers.fixture("fixtures/sender.json"), Sender::class.java))
        Assert.assertEquals(expected, MAPPER.writeValueAsString(tag))
    }

    @Test
    fun deserializeFromJson() {
        val tag = Sender(id = 1, name = "sender1")
        Assert.assertEquals(MAPPER.readValue(FixtureHelpers.fixture("fixtures/sender.json"), Sender::class.java), tag)
    }

    companion object {
        // https://github.com/FasterXML/jackson-module-kotlin
        private val MAPPER = jacksonObjectMapper()
    }
}